sdk/storage/Azure.Storage.DataMovement/src/CommitChunkHandler.cs

```diff
@@ -119,25 +114,37 @@ public async ValueTask DisposeAsync()
             DipsoseHandlers();
         }
 
-        public void DipsoseHandlers()
+        private void DipsoseHandlers()
         {
             if (_transferType == TransferType.Sequential)
             {
-                _commitBlockHandler -= QueueBlockEvent;
+                _commitBlockHandler -= SequentialBlockEvent;
             }
-            _commitBlockHandler -= CommitBlockEvent;
+            _commitBlockHandler -= ConcurrentBlockEvent;
         }
 
-        private async Task CommitBlockEvent(StageChunkEventArgs args)
+        private async Task ConcurrentBlockEvent(StageChunkEventArgs args)
         {
-            if (args.Success)
+            try
             {
-                // Let's add to the channel, and our notifier will handle the chunks.
-                await _stageChunkChannel.Writer.WriteAsync(args, _cancellationToken).ConfigureAwait(false);
+                if (args.Success)
+                {
+                    // Let's add to the channel, and our notifier will handle the chunks.
+                    await _stageChunkChannel.Writer.WriteAsync(args, _cancellationToken).ConfigureAwait(false);
```

Comment:
Could this actually throw an exception? Just writing to the channel isn't performing the operations right? I guess a cancellation exception?

Reply:
It can throw an `OperationCancelledException`. In which gets filtered down the line.